internal/pkg/bootstrap/handlers/database/database.go

```diff
@@ -104,7 +105,8 @@ func (d Database) BootstrapHandler(
 	var dbClient dbInterfaces.DBClient
 	for startupTimer.HasNotElapsed() {
 		var err error
-		dbClient, err = d.newDBClient(loggingClient)
+		credentials, err := db.GetDatabaseCredentials(dic, d.database.GetDatabaseInfo()["Primary"])
```

Comment:
You're still passing the container around.  Pull out the required dependencies and pass them explicitly.  Results in the called function/method being infinitely more testable.

Reply:
I was trying to not duplicate code, but it seems like we are going to have to here since we will need to check if security is enabled before getting the `SecretClient` out of the container. At that point the `GetDatabaseCredentials` helper function is useless. I will just remove that helper function.

Reply:
Missing check for err != nil